src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -39,98 +40,234 @@ private ExpressionSyntax ParseInterpolatedStringToken()
+                // line, making sure its indentation is correct.
+
+                SyntaxDiagnosticInfo? indentationError = null;
+                while (currentIndex < text.Length)
                 {
-                    for (int i = 0; i < interpolations.Count; i++)
+                    var lineStartPosition = currentIndex;
+
+                    // Only bother reporting a single indentation error on a text chunk.
+                    if (error == null)
                     {
-                        var interpolation = interpolations[i];
+                        currentIndex = SkipWhitespace(text, currentIndex);
+                        var currentLineWhitespace = text[lineStartPosition..currentIndex];
 
-                        // Add a token for text preceding the interpolation
-                        var text = originalText[new Range(
-                            i == 0 ? openQuoteRange.End : interpolations[i - 1].CloseBraceRange.End,
-                            interpolation.OpenBraceRange.Start)];
-                        if (text.Length > 0)
+                        if (!currentLineWhitespace.StartsWith(indentationWhitespace))
                         {
-                            builder.Add(SyntaxFactory.InterpolatedStringText(MakeInterpolatedStringTextToken(text, kind)));
+                            // We have a line where the indentation of that line isn't a prefix of indentation
+                            // whitespace.
+                            //
+                            // If we're not on a blank line then this is bad.  That's a content line that doesn't start
+                            // with the indentation whitespace.  If we are on a blank line then it's ok if the whitespace
+                            // we do have is a prefix of the indentation whitespace.
+                            var isBlankLine = (currentIndex == text.Length && last) || (currentIndex < text.Length && SyntaxFacts.IsNewLine(text[currentIndex]));
+                            var isLegalBlankLine = isBlankLine && indentationWhitespace.StartsWith(currentLineWhitespace);
+                            if (!isLegalBlankLine)
+                            {
+                                indentationError ??= MakeError(
+                                    lineStartPosition,
+                                    width: currentIndex - lineStartPosition,
+                                    ErrorCode.ERR_LineDoesNotStartWithSameWhitespace);
+                            }
                         }
-
-                        builder.Add(ParseInterpolation(this.Options, originalText, interpolation, kind));
                     }
 
-                    // Add a token for text following the last interpolation
-                    var lastText = originalText[new Range(interpolations[^1].CloseBraceRange.End, closeQuoteRange.Start)];
-                    if (lastText.Length > 0)
```

Comment:
Why remove the length checks that we had before, any time we call `builder.Add(...)`?
I'm surprised we're not getting nullability warning from `builder.Add(makeContent(...))` in new code.

Reply:
we actaully are getting the nullability warnings.  it's suppressed.  I can actually update the annotation on .Add and doc that it will no-op on null.